nixos/modules/services/network-filesystems/glusterfs.nix

```diff
@@ -52,6 +52,8 @@ in
 
       preStart = ''
         install -m 0755 -d /var/log/glusterfs
+        mkdir -p /var/lib/glusterd/hooks/
+        ${rsync}/bin/rsync -a ${glusterfs}/var/lib/glusterd/hooks/ /var/lib/glusterd/hooks/
```

Comment:
@FRidh I do not know what gluster's assumptions are about these files being writable.

I believe the right way to do this is to have gluster not expect these files that are required for operation in `/var`, but inside its install prefix instead. People on the `#gluster` IRC channel seem to share this opinion.

I have filed an upstream bug about moving those files: https://bugzilla.redhat.com/show_bug.cgi?id=1452761